src/nvim/main.c

```diff
@@ -1444,6 +1444,18 @@ static void command_line_scan(mparm_T *parmp)
       ga_grow(&global_alist.al_ga, 1);
       char *p = xstrdup(argv[0]);
 
+      // On Windows expand "~\" or "~/" prefix in file names to profile directory.
+#ifdef MSWIN
+      if (*p == '~' && (p[1] == '\\' || p[1] == '/')) {
+        const char *profile_dir = vim_getenv("HOME");
+        size_t size = strlen(profile_dir) + strlen(p);
+        char *tilde_expanded = xmalloc(size);
+        snprintf(tilde_expanded, size, "%s%s", profile_dir, p + 1);
+        xfree(p);
```

Comment:
`profile_dir` is allocated, so it needs to be freed as well.
```suggestion
        xfree(p);
        xfree(profile_dir);
```